crypto/cms/cms_smime.c

```diff
@@ -87,14 +88,13 @@ static int check_content(CMS_ContentInfo *cms)
 
 static void do_free_upto(BIO *f, BIO *upto)
 {
-    if (upto) {
+    if (upto != NULL) {
         BIO *tbio;
         do {
             tbio = BIO_pop(f);
             BIO_free(f);
             f = tbio;
-        }
-        while (f && f != upto);
+        } while (f != upto);
```

Comment:
Is there a reason you wanna walk past NULL?

Reply:
wise guy. :)  fixed.